compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1973,6 +1974,34 @@ object messages {
     val explanation = "A sealed class or trait can only be extended in the same file as its declaration"
   }
 
+  case class SymbolHasUnparsableVersionNumber(symbol: Symbol, migrationMessage: String)(implicit ctx: Context)
+  extends Message(SymbolHasUnparsableVersionNumberID) {
+    val kind = "Syntax"
+    val msg = hl"${symbol.showLocated} has an unparsable version number: $migrationMessage"
+    val explanation =
+      hl"""$migrationMessage
+          |
+          |The symbol is marked with ${"@migration"} indicating it has changed semantics
```

Comment:
You should not talk about symbols as this is a compiler internal concept, the user should not know about them. You should probably replace it with `${symbol.showLocated}` or `${symbol.show}`.